src/Profiles.ts

```diff
@@ -1351,32 +816,26 @@ export class Profiles {
         const OrigProfileInfo = this.loadedProfile.profile;
         const NewProfileInfo = ProfileInfo.profile;
 
-        const profileArray = Object.keys(this.loadedProfile.profile);
+        const profileArray = Object.keys(NewProfileInfo);
         for (const value of profileArray) {
-            if (value === "user" || value === "password") {
-                if (!rePrompt) {
-                        OrigProfileInfo.user = NewProfileInfo.user;
-                        OrigProfileInfo.password = NewProfileInfo.password;
-                }
-            } else {
-                OrigProfileInfo[value] = NewProfileInfo[value];
-            }
-
+            OrigProfileInfo[value] = NewProfileInfo[value];
+            if (NewProfileInfo[value] == null) { delete OrigProfileInfo[value]; }
         }
 
-        // Using `IUpdateProfileFromCliArgs` here instead of `IUpdateProfile` is
-        // kind of a hack, but necessary to support storing secure credentials
-        // until this is fixed: https://github.com/zowe/imperative/issues/379
-        const updateParms: IUpdateProfileFromCliArgs = {
+        const updateParms: IUpdateProfile = {
             name: this.loadedProfile.name,
-            merge: true,
-            // profile: OrigProfileInfo as IProfile
-            args: OrigProfileInfo as any
+            merge: false,
+            profile: OrigProfileInfo as IProfile
         };
         try {
-            (await this.getCliProfileManager(this.loadedProfile.type)).update(updateParms);
+            const updatedProfile = await (await this.getCliProfileManager(this.loadedProfile.type)).update(updateParms);
+            return updatedProfile.profile;
         } catch (error) {
-            vscode.window.showErrorMessage(error.message);
+            // When no password is entered, we should silence the error message for not providing it
+            // since password is optional in Zowe Explorer
+            if (!error.message.includes("Must have user & password OR base64 encoded credentials")) {
```

Comment:
I would prefer not to have this. Credentials are optional in CLI as well so maybe we need to investigate further why this message keeps on prompting. We need to handle those scenarios better.

Reply:
This has been changed, we need to test this again :V